tests/unit/io/Cereal_unittest.cc

```diff
@@ -0,0 +1,71 @@
+#include <shogun/lib/common.h>
+#include <shogun/base/Parameter.h>
+#include <gtest/gtest.h>
+#include <../tests/unit/io/CerealObject.h>
+
+#include <shogun/lib/SGVector.h>
+#include <cereal/archives/json.hpp>
+#include <fstream>
+
+
+using namespace shogun;
+
+TEST(Cereal, Json_SGVector_FLOAT64_save_load_pair)
+{
+	const index_t size = 5;
+	SGVector<float64_t> a(size), b;
+	a.range_fill(1.0);
+
+	{
+		std::ofstream os("cereatltest_sgvector.cereal");
+		cereal::JSONOutputArchive archive(os);
+		archive(a);
+	}
+
+	{
+		std::ifstream is("cereatltest_sgvector.cereal");
+		cereal::JSONInputArchive archive(is);
+		archive(b);
+	}
+
+	for (index_t i = 0; i < size; i++)
+		EXPECT_NEAR(a[i], b[i], 1E-15);
+}
+
+TEST(Cereal, Json_AnyObject_save_and_load_methods)
+{
+	SGVector<float64_t> vec_a(5), vec_b(5);
+	vec_a.range_fill(0);
+	vec_b.range_fill(1);
+	Any a(vec_a), b(vec_b);
+
+	{
+		std::ofstream os("cereatltest_save_anyobject.cereal");
+		cereal::JSONOutputArchive archive(os);
+		archive(a);
+	}
+
+	{
+		std::ifstream is("cereatltest_save_anyobject.cereal");
+		cereal::JSONInputArchive archive(is);
+		archive(b);
+	}
+
+	EXPECT_TRUE(a==b);
+}
+
+TEST(Cereal, Json_CerealObject_save_and_load_methods)
+{
+	CCerealObject obj_save, obj_load;
+	SGVector<float64_t> A(5), B;
+	A.range_fill(0);
+
+	obj_save.set<SGVector<float64_t>>("test_vector", A);
+	obj_save.saveJSON("cereatltest_save_cerealobject.cereal");
+
+	obj_load.loadJSON("cereatltest_save_cerealobject.cereal");
+	B = obj_load.get<SGVector<float64_t>>("test_vector");
+
```

Comment:
Can you explain what you mean by this?

 * de-serializatzion directly allocates member variable memory and loads the data there
 * setters (if exist) might allow to change members
 * otherwise no access